### ISSUE TITLE
OnDropItemRequested Change

### ISSUE THREAD

**Issue by rube200** [SmartlyDressedGames/Unturned-3.x-Community#1444]
Original:
public delegate void DropItemRequestHandler(PlayerInventory inventory, Item item, ref bool shouldAllow);

Change:
public delegate void DropItemRequestHandler(PlayerInventory inventory, ItemJar item, ref bool shouldAllow);

I think if instead of passing the Item it should be ItemJar it opened up a wider range of possibilities,
I, for example, do not want items dropped by players to appear on the floor, so I have to handle this event and also onServerSpawningItemDrop, but if the ItemJar was passed, I could delete it directly from the inventory

**Comment by Edsparr** [comment]
You can find the ItemJar from the inventory already, you just need to compare them. I would agree though this change would be cleaner, though it would break compatibility.

TL;DR Already possible and cost of changing might be too expensive

**Comment by rube200** [comment]
@Edsparr First: yes, it is true that you can get the ItemJar from the inventory, but this causes unnecessary processing from the moment @SDGNelson can just delete '.item'.
```
if (this.onDropItemRequested != null)
    this.onDropItemRequested(this, itemJar.item, ref allowManualDrop);
```
Second: This event is only used by plugins, plugins can be changed and fixed. Regarding compatibility, @SDGNelson  is always breaking the game, luckily it doesn't happen that often, but if it is really necessary, I think there may be two events as it happens in other parts of the code.

**Comment by erik7302** [comment]
maybe what nelson can do is add it under other name and mark the old one with [Obsolete]
